preview/MsixCore/msixmgr/CryptoProvider.cpp

```diff
@@ -0,0 +1,191 @@
+#include <Windows.h>
+#include <WinCrypt.h>
+
+#include <BcryptLibrary.hpp>
+#include "GeneralUtil.hpp"
+#include <CryptoProvider.hpp>
+#include <TraceLoggingProvider.h>
+#include "MsixTraceLoggingProvider.hpp"
+#include <assert.h>
+
+namespace MsixCoreLib
+{
+    CryptoProvider::CryptoProvider()
+    {
+        this->providerHandle = NULL;
+        this->hashHandle = NULL;
+        this->digest.bytes = NULL;
+        this->digest.length = 0;
+    }
+
+    CryptoProvider::~CryptoProvider()
+    {
+        Reset();
+    }
+
+    void CryptoProvider::Reset()
+    {
+        this->digest.bytes = NULL;
+        this->digest.length = 0;
+
+        if (NULL != this->hashHandle)
+        {
+            if (!SUCCEEDED(BcryptLibrary::BCryptDestroyHash(this->hashHandle)))
+            {
+                return;
+            }
+            this->hashHandle = NULL;
+        }
+
+        if (NULL != this->providerHandle)
+        {
+            if (!SUCCEEDED(BcryptLibrary::BCryptCloseAlgorithmProvider(this->providerHandle, 0)))
+            {
+                return;
+            }
+            this->providerHandle = NULL;
+        }
+    }
+
+    HRESULT CryptoProvider::Create(
+        _Outptr_ CryptoProvider** provider)
+    {
+        RETURN_IF_FAILED(BcryptLibrary::Load());
+
+        std::unique_ptr<CryptoProvider> cp(new APPXCOMMON_NEW_TAG CryptoProvider());
+        //RETURN_IF_NULL_ALLOC(cp);

```

Comment:
>RETURN_IF_NULL_ALLOC [](start = 10, length = 20)

don't just comment out the code, replace it with the appropriate check or add the macro

Reply:
Replaced with correct check and error handling.